src/tss2-fapi/ifapi_helpers.c

```diff
@@ -2434,85 +2434,79 @@ write_curl_buffer_cb(void *contents, size_t size, size_t nmemb, void *userp)
-        goto_error(r, TSS2_FAPI_RC_GENERAL_FAILURE, "Curl easy setopt",
-                   cleanup);
+    CURL *curl = curl_easy_init();
+    if (!curl) {
+        LOG_ERROR("curl_easy_init failed");
+        goto out_global_cleanup;
     }
-    if (CURLE_OK != curl_easy_setopt(curl_handle, CURLOPT_WRITEFUNCTION,
-                                     write_curl_buffer_cb)) {
-        goto_error(r, TSS2_FAPI_RC_GENERAL_FAILURE, "Curl easy setopt",
-                   cleanup);
+
+    rc = curl_easy_setopt(curl, CURLOPT_URL, url);
+    if (rc != CURLE_OK) {
+        LOG_ERROR("curl_easy_setopt for CURLOPT_URL failed: %s",
+                curl_easy_strerror(rc));
+        goto out_easy_cleanup;
     }
-    if (CURLE_OK != curl_easy_setopt(curl_handle, CURLOPT_WRITEDATA,
-                                     (void *)&curl_buffer)) {
-        goto_error(r, TSS2_FAPI_RC_GENERAL_FAILURE, "Curl easy setopt",
-                   cleanup);
+
+    rc = curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION,
+                          write_curl_buffer_cb);
+    if (rc != CURLE_OK) {
+        LOG_ERROR("curl_easy_setopt for CURLOPT_URL failed: %s",
+                curl_easy_strerror(rc));
+        goto out_easy_cleanup;
     }
-    if (CURLE_OK != curl_easy_setopt(curl_handle, CURLOPT_USERAGENT,
-                                     "libcurl-agent/1.0")) {
-        goto_error(r, TSS2_FAPI_RC_GENERAL_FAILURE, "Curl easy setopt",
-                   cleanup);
+
+    rc = curl_easy_setopt(curl, CURLOPT_WRITEDATA,
+                          (void *)&curl_buffer);
+    if (rc != CURLE_OK) {
+        LOG_ERROR("curl_easy_setopt for CURLOPT_URL failed: %s",
+                curl_easy_strerror(rc));
+        goto out_easy_cleanup;
     }
+
     if (LOGMODULE_status == LOGLEVEL_TRACE) {
-        if (CURLE_OK != curl_easy_setopt(curl_handle, CURLOPT_VERBOSE, 1L)) {
+        if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L)) {
             LOG_WARNING("Curl easy setopt verbose failed");
         }
     }
 
-    /* Receive the certificate */
-    if (CURLE_OK != curl_easy_perform(curl_handle)) {
-        goto_error(r, TSS2_FAPI_RC_GENERAL_FAILURE, "Curl easy setopt",
-                   cleanup);
+    rc = curl_easy_perform(curl);
+    if (rc != CURLE_OK) {
+        LOG_ERROR("curl_easy_perform() failed: %s", curl_easy_strerror(rc));
+        goto out_easy_cleanup;
     }
-    LOG_TRACE("%zu bytes of certificate retrieved\n", curl_buffer.size);
 
     *buffer = curl_buffer.buffer;
     *buffer_size = curl_buffer.size;
-    if (curl_handle)
-        curl_easy_cleanup(curl_handle);
-    curl_global_cleanup();
-    return r;
 
-cleanup:
-    if (curl_handle)
-        curl_easy_cleanup(curl_handle);
+    ret = 0;
+
+out_easy_cleanup:
+    if (ret != 0)
+        free(curl_buffer.buffer);
+    curl_easy_cleanup(curl);
+out_global_cleanup:
     curl_global_cleanup();
-    free(curl_buffer.buffer);
-    return r;
+out_memory:
+    return ret;
 }
+
```

Comment:
Please pay attention to these empty lines at the EOF because it looks like it upsets the CI:
https://travis-ci.org/tpm2-software/tpm2-tss/jobs/654483762#L2446